nextcord/ext/commands/core.py

```diff
@@ -1177,13 +1177,16 @@ async def can_run(self, ctx: Context) -> bool:
                 if local_check is not None:
                     ret = await nextcord.utils.maybe_coroutine(local_check, ctx)
                     if not ret:
+                        ctx.command = original
                         return False
 
             predicates = self.checks
             if not predicates:
                 # since we have no checks, then we just return True.
+                ctx.command = original
                 return True
 
+            ctx.command = original
             return await nextcord.utils.async_all(predicate(ctx) for predicate in predicates)  # type: ignore
```

Comment:
Would this not change the behavior of these predicates? I'm not entirely sure how this functions but it appears like this is a logic change and you were possibly aiming for something more in line with:
```python
result =  await nextcord.utils.async_all(predicate(ctx) for predicate in predicates)  # type: ignore
ctx.command = original
return result
```

Which I think would keep the same behavior, while also resetting the command after its been used by the help command. However, this is an untested theory.

Reply:
it seems as though this is more complicated than i thought, doing it this way seems to cause the command invoked to run, maybe the command should be `copy.copy()`'d before dispatching so the original one is dispatched?

Reply:
`copy.copy` and this suggestion has been applied